### ISSUE TITLE
Interface stops responding after a while in Safari

### ISSUE THREAD

**Issue by itwasinthetubes** [RainLoop/rainloop-webmail#421]
On Safari Mac (Yosemite)
everything works fine for a while, but then all buttons stop being interactive. This happens after browsing the web away from rainloop in other tabs and then returning to check emails.

I have not been able to find the exact case when it happens but it occurs multiple times in one day of continuos use. The only solution is to reload the page to regain interactivity.

This has been an issue even after updating rainloop the last 3-4 times and multiple versions of safari and yosemite beta releases (Latests releases from Mac Developer Program). At least in the last 6 months this has happened.

Anyone else have this issue?

**Comment by hachre** [comment]
This is a new feature in Safari that "pauses" websites and JS execution in them. When you go back to the tab the JS is resumed. However connections (even WebSockets) get interrupted and instantly run the  appropriate disconnection events to which your app has to react and then reconnect. 

There is no event to listen to to know when your website has been paused or resumed as far as I know but you can basically create one yourself by running a timer lets say every 1000 msec and comparing a stored millisecond date value to the current value. If the values are significantly more than 1000 msec apart your website was sleeping and you can trigger whatever you need to make the JS run again.

**Comment by caudurow3** [comment]
So, by saving energy of hidden tabs with Safari they crash JS apps? Did not know about this. 

As a side note, I can confirm the problem does not occur in Firefox v 35 on Mac Yosemite. 

But as hachre explains above, that sounds like the culprit. Shame, as rainloop works well apart from that on Safari. I have switched to FF for the time being.

**Comment by hachre** [comment]
Yeah, I don't know what the feature is called exactly but it came with the new Safari in Yosemite. It a power saving feature. It happens when you drag a window on top of the Safari window, or when switching to another tab. 

Basically when the website content is visually hidden through any means Safari _might_ stop it, but only if there is a certain amount of memory pressure and other factors that Apple didn't really get into.

**Comment by RainLoop** [comment]
Does you use RainLoop with ownCloud ?

**Comment by caudurow3** [comment]
Yes - with owncloud integration, but also separately - domain.com/rainloop

**Comment by RainLoop** [comment]
Right now there is a problem with owncloud integration #445 . Maybe it's the same problem.

**Comment by caudurow3** [comment]
It does not sound like the same problem...
After switching to other browsers the problem does not happen. 

I also didn't have the spike in CPU, just a complete lack of interactivity from all buttons in safari (the page became static - you could still select some elements to copy and paste, but all buttons stop responding).

**Comment by megadr01d** [comment]
Try to open Finder, call **Get Info** on Safari and disable **App Nap**. This is the "new technology" started in Mavericks to pause applications in the background. I found it caused some system freezing in an app or two so I disabled App Nap on all apps that support this. For the record, I also enabled **Open in 32bit** on all apps that support it just because it's enough and uses less memory for each app.
